pkg/importer/dataStream.go

```diff
@@ -361,7 +375,11 @@ func (d *DataStream) fileFormatSelector(hdr *image.Header) (err error) {
 		r, d.Size, err = d.qcow2NopReader(hdr)
 		d.qemu = true
 	case "tar":
-		r, d.Size, err = d.tarReader()
+		if d.content == common.ImporterContentArchive {
+			r, d.Size, err = d.mulFileTarReader()
```

Comment:
would be good to abstract the d.mulFileTarReader and the tarReader functions into a single wrapper function that takes d.content and let it figure out which one to call.

Reply:
Good idea, thanks!